plugins/woocommerce/includes/wc-update-functions.php

```diff
@@ -2594,3 +2595,46 @@ function wc_update_750_disable_new_product_management_experience() {
 		update_option( 'woocommerce_new_product_management_enabled', 'no' );
 	}
 }
+
+/**
+ * Delete stray order records in both posts table and the orders table:
+ *
+ * - Posts of type "shop_order_placeholder" with no matching order in the orders table.
+ * - Records of type "shop_order_refund" with a zero parent order id in both tables.
+ *
+ * @return void
+ */
+function wc_update_770_delete_stray_order_records() {
```

Comment:
But the records that get deleted by this migration are unusable anyway: refunds with no parent order id (so there's no way to know which order do they belong and thus aren't used for anything), and placeholder records for which there's no matching order in `wp_wc_orders` (so they aren't "placeholding" anything).

Reply:
Not sure about that, we can't say for sure if we created those refunds (instead of some plugin, or perhaps an import process). It's going towards very edge cases, but then there is also no harm in just keeping these records. wdyt?
Although, the placeholders records are fine, since we know we added them.

Reply:
I must have been doing something wrong before, but now I see that I can use `wc_get_order` for the child orders and I don't get the "Invalid parent ID" errors. So I have simplified the code accordingly (including removing the new `verify_parent_id` methods).

Reply:
Implemented in https://github.com/woocommerce/woocommerce/pull/37050/commits/bb6645d4acd8e3b846c58b8a8ecc7ccf89cd4c1b